x/nft/username/handler.go

```diff
@@ -249,3 +249,71 @@ func findActor(h tokenHandler, ctx weave.Context, t Token) weave.Address {
 	}
 	return nil
 }
+
+type AddRemoveApprovalHandler struct {
+	tokenHandler
+}
+
+func (h *AddRemoveApprovalHandler) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	if _, err := h.validate(ctx, tx); err != nil {
+		return res, err
+	}
+	res.GasAllocated += createUsernameCost
+	return res, nil
+}
+
+func (h *AddRemoveApprovalHandler) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.DeliverResult, error) {
+	var res weave.DeliverResult
+	msg, err := h.validate(ctx, tx)
+	if err != nil {
+		return res, err
+	}
+
+	o, t, err := loadToken(h.tokenHandler, store, msg.GetId())
+	if err != nil {
+		return res, err
+	}
+
+	actor := findActor(h.tokenHandler, ctx, t)
+	if actor == nil {
+		return res, errors.ErrUnauthorized()
+	}
+
+	switch v := interface{}(msg).(type) {
+	case *AddApprovalMsg:
+		base := v.Base
+
+		err = t.Approvals().Grant(base.Action, base.Address, base.Options, 0)
+		if err != nil {
+			return res, err
+		}
+	case *RemoveApprovalMsg:
```

Comment:
probably a stupid question, but Go doesnt require a default case for switch?

Reply:
well spotted. There is no requirement for a default case in the language